### PR TITLE
Signing APDU

### PR DESCRIPTION
**Pull request by jonored**
Includes the provide public key APDU and stubbed wallet ID APDU.

This implements only the fallback "sign hash" functionality so far.

### CODE REVIEW

**Comment by jonored** [comment]
App crashes when the display cycles; still debugging.

File: release.nix [hunk 1]
```diff
@@ -1,7 +1,6 @@
 let
   ledger-app-nervos = import ./. {};
-in {
   analysis-nanos = ledger-app-nervos.clangAnalysis.s;
   release-nanos = ledger-app-nervos.nano.s.release;
   ckb-cli = (import ./nix/dep/ckb-cli {});
-}
+in [ analysis-nanos release-nanos ckb-cli ]
```
Comment (matthewbauer) [inline release.nix]: I don't think this is necessary?

File: README.md [hunk 2]
```diff
@@ -33,16 +33,34 @@ Testing need to be done through Ledger Live.
...
+```
+
+The test suite currently does not test any rejections, so just accept every
+prompt that happens.
+
 ## Manual Testing ##
 
 Not everything is coverd by the automated tests, so manual testing is
-also necessary. This enables end-to-end testing of the ckb 
+also necessary. This enables end-to-end testing of the ckb
```
Comment (matthewbauer) [inline README.md]: (should probably be a period here).

File: src/blockchain.h [hunk 3]
```diff
@@ -0,0 +1,1134 @@
+// Generated by Molecule 0.5.0
```
Comment (matthewbauer) [inline src/blockchain.h]: With this, you can probably remove extensions.c and blockchain.c.

**Pull request merged**